src/s3Client.ts

```diff
@@ -1,11 +1,22 @@
 import { S3 } from '@aws-sdk/client-s3';
+import { requireEnv } from 'require-env-variable';
```

Comment:
This triggered a series of thoughts, so I'm writing them down:

1. I bet there are other places we should be more explicitly requiring env vars (doesn't have to be in this PR)
2. What does dotenv get us / should we still use dotenv (I think dotenv is providing the ability to autoload a `.dotenv` file in the event that env vars aren't automatically sourced, yes?  Do we care enough about that to warrant having two env var dependencies?)
3. I suppose a way to accomplish this within dotenv would be to create a class dedicated to environment variable management, and access env vars from that class.

None of these thoughts warrant a change to this PR necessarily, but I wanted to raise them.

I am comfortable with this approach as written, though I could imagine there being a benefit from a custom class that defines (and validates) environment variables even more robustly.

Reply:
> I bet there are other places we should be more explicitly requiring env vars

I actually think this is the last of them! `git grep process.env` is a pleasantly short and conditional.

> What does dotenv get us / should we still use dotenv

It is a convenience for developers, to automatically load the contents of a `.env` file. I am increasingly of the opinion that such convenience support should not live in the repo, but rather be the responsibility of the devs, via [direnv](https://direnv.net/) or another way of managing environment variables. However, I do not believe so strongly enough to advocate for removing existing, working setups, so for now my answer to the second question is "eh, sure".

> create a class dedicated to environment variable management

We could. I think the main benefit would be consolidating all the environment variables in one place; currently, if say `OPENAPI_DOCS_AUTH_CLIENT_ID` and `S3_ACCESS_KEY_ID` were both missing, only one would be reported because they're in separate calls to `requireEnv`. I'm not sure it's worth the work and conceptual overhead, but could probably be convinced.

Reply:
> I actually think this is the last of them!

I *may* be missing it but it looks like this doesn't get at the env var in https://github.com/PhilanthropyDataCommons/service/blob/main/src/logger.ts right?

> It is a convenience for developers

Ahhhh -- then... I do think we should plan to get rid of it especially due to the awkwardness documented in `index.ts` around import orders mattering (but I'll open an issue for it).

(the class manager convo I'm happy to let sit as a musing in a PR thread and not carry it forward at all!)

Reply:
(as noted on call -- this is not required so requiring wouldn't be good lulz)

Reply:
Captured in https://github.com/PhilanthropyDataCommons/service/issues/640!